### PR TITLE
Simplify README and point to new docs

### PR DESCRIPTION
**Pull request by mfreed**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,325 +5,20 @@ time-series data. It is engineered up from PostgreSQL, providing automatic
...
+Additional documentation can be found [on our docs
+website](http://docs.timescale.com/), including the following:
 
-_NOTE: Currently, upgrading to new versions requires a fresh install._
+- [Why use TimescaleDB?](http://docs.timescale.com/introduction)
+- [Migrating from PostgreSQL](http://docs.timescale.com/getting-started/setup/migrate-from-postgresql)
+- [Writing data](http://docs.timescale.com/api#insert)
+- [Querying and data analytics](http://docs.timescale.com/api#select)
+- [Tutorials and sample data](http://docs.timescale.com/tutorials)
+- [Contributing to TimescaleDB](https://github.com/timescale/timescaledb/blob/master/CONTRIBUTING.md)
```
Comment (solugebefola) [inline README.md]: This one isn't really on the docs site...

Comment (mfreed) [inline README.md]: There was a request for the link.  Do you prefer pulling it out as a separate sentence?

Comment (solugebefola) [inline README.md]: yeah, I think so.

**Comment by akulkarni** [comment]
LGTM

**Comment by mfreed** [comment]
There were two key reasons for moving this all to the docs:

- So we have *one* place to keep things up-to-date, so that instructions don't get out of sync between the docs and github, or slightly vary and lead to confusion.  

   This was already happening before between github and our old docs, and we didn't have a single place to point people to.  When installing, we've seen users needing to jump between the github README and docs -- it was very confusing.

- To provide users with a clear "choose-your-own-adventure" style of instructions.  We now maintain a number of different ways for installing TimescaleDB, all with (slightly to widely) varying instructions.  

    It was getting very unwieldy to have this all on one page, and also gave users the appearance that things were much more difficult than they are.  

    The designed docs now give a very nice flow that are relevant to each user's particular OS & preferred method of install.  If this is replicated in github, however, I think users will often miss the better install docs -- or be confused about which they should use -- and instead just slog it out with the github README.  This problem would only get worse when we add instructions for various cloud providers, etc.

**Comment by RobAtticus** [comment]
I think I'm in agreement with @erimatnor about install instructions. One thought I had was something simply like:

*MacOs*:
```bash
brew tap timescale/tap
brew install timescaledb
```
*Ubuntu* (16.04+)
```bash
sudo add-apt-repository ppa:timescale/timescaledb-ppa
sudo apt-get update
sudo apt install timescaledb
```
*Fedora/CentOS*
```bash
sudo yum install -y https://download.postgresql.org/pub/repos/yum/9.6/fedora/fedora-24-x86_64/pgdg-fedora96-9.6-3.noarch.rpm
wget https://timescalereleases.blob.core.windows.net/rpm/timescaledb-0.0.11~beta-0.x86_64.rpm
sudo yum install timescaledb
```
*Docker*
```bash
docker pull timescale/timescaledb:latest
```

Then a link to our docs for further instructions including updating postgresql.conf, etc

**Comment by akulkarni** [comment]
Normally I'd agree with @erimatnor and @RobAtticus, but our installation instructions are so complicated (largely b/c of PostgreSQL), that listing them in a markdown file seems insufficient (ie it's confusing).

I suggest we give this other approach a try and see how that affects the installation experience (using both qualitative and quantitative metrics).

**Comment by erimatnor** [comment]
To clarify my position, I think the README should provide the complete set of instruction for building, installing and basic usage.

Include:
- Prerequisites (i.e., Postgres + version, development headers/packages)
- How to build this extension from source (`make`)
- How to install the built extension (`make install`)
- Basic usage: `create extension` + `SELECT create_hypertable()`
- Link to web docs for full instructions (tutorials, postgres + extension installation steps, etc.)

Leave out (link to webdocs):
- Installing postgres
- Installing pre-built packages
- Recommended settings, tuning, etc.
- Advanced usage: retention, indexes, etc.
- Tutorials

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -87,13 +87,22 @@ our [Style Guide](docs/StyleGuide.md).
...
+relevant addition to or modification of the test suite.
 
 Please check that the full test suite (including your test additions
 or changes) passes successfully on your local machine before you
-open a pull request.  All submitted pull requests are also automatically
+open a pull request.
+
+If you are running locally:
+```bash
+make installcheck
```
Comment (erimatnor) [inline CONTRIBUTING.md]: Slightly weird to have `make` commands for testing, but not for building.

Comment (mfreed) [inline CONTRIBUTING.md]: This is instructions for how people can contribute.

**Comment by akulkarni** [comment]
Approving for now, but let's revisit asap and add whatever is missing.

**Comment by mfreed** [comment]
Yes, let's review also after we update main links from website.  

For example, right now splash does not point to download in docs, so users are naturally all driven to github for download (and the above would them seem to "recommend" git clone as the preferred method).  

This also might lead to some other issues, e.g., if users clone the master branch rather than a tag set, auto-upgrades of the DB (may) become more difficult.

So, let's revisit.

**Pull request merged**